### PR TITLE
chore(release-notes): parse Renovate bot release notes for allowed packages

### PR DESCRIPTION
**Pull request by binoy14**
## Summary

- Add `parseRenovateReleaseNotes()` to extract structured changelog content from Renovate bot PR bodies (which use `<details>` HTML blocks that don't survive portable text conversion)
- Only parses packages in a configurable allowlist (`RENOVATE_RELEASE_NOTES_PACKAGES`, currently `@sanity/cli`)
- Filters out hidden sections (Dependencies, Styles, Chores, etc.) and strips commit hash/PR ref noise from list items
- Bot PRs with successfully parsed release notes are no longer auto-excluded from the changelog

## Test plan

- [x] New unit tests for the parser (11 test cases covering allowlist filtering, section filtering, noise stripping, edge cases)
- [x] Existing `extractReleaseNotes` tests still pass (no regressions)
- [ ] Verify on a real Renovate PR (e.g. #12405) that the parsed output contains expected bug fix/feature descriptions

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: packages/@repo/release-notes/src/utils/parseRenovateReleaseNotes.ts [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+ * Strip trailing commit hashes like `(0b660b9)` and PR refs like `(#661)` from list items,
+ * and remove dependency-scoped items (e.g. `- **deps:** ...`) that aren't meaningful for release notes.
+ */
+function cleanChangelogItems(markdown: string): string {
+  return (
+    markdown
+      // Remove dependency-scoped list items
+      .replace(/^\s*-\s+\*\*deps:\*\*.*$/gm, '')
+      // Strip PR refs and commit hashes from remaining items
+      .replace(/^(\s*-\s+.+?)\s*\(\[#\d+]\([^)]*\)\)\s*\(\[[0-9a-f]+]\([^)]*\)\)\s*$/gm, '$1')
```
Comment (rexxars) [inline packages/@repo/release-notes/src/utils/parseRenovateReleaseNotes.ts]: `conventional-changelog` _can_ produce items with only one of the two, or in reversed order (I personally don't like the commit hashes, so sometimes disable them), so maybe make them optional?

```suggestion
      .replace(/^(\s*-\s+.+?)\s*(?:\(\[#\d+]\([^)]*\)\)\s*)?(?:\(\[[0-9a-f]+]\([^)]*\)\)\s*)?$/gm, '$1')
```
